components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -2049,6 +2068,26 @@ private Map<String, String> getApplicationDetails(AuthenticationContext context)
         return applicationDetailsMap;
     }
 
+    /**
+     * Extract query param scopes from a given url.
+     *
+     * @param url Given url.
+     * @return Extracted scopes as a String.
+     */
+    protected String extractScopesFromURL(String url) throws UnsupportedEncodingException {
+
```

Comment:
Shall we extract the query params first from URL and them execute the next logic on top of that?

If the scope is the first query param (`<URL>?scope=a b c&...`), I think the following logic does not recognize the scopes